OracleDatabase/SingleInstance/dockerfiles/21.3.0/runOracle.sh

```diff
@@ -0,0 +1,269 @@
+   mv $ORACLE_BASE_HOME/network/admin/sqlnet.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+   mv $ORACLE_BASE_HOME/network/admin/listener.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+   mv $ORACLE_BASE_HOME/network/admin/tnsnames.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+   mv $ORACLE_HOME/install/.docker_* $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+
+   # oracle user does not have permissions in /etc, hence cp and not mv
+   cp /etc/oratab $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+   
+   symLinkFiles;
+}
+
+########### Symbolic link DB files ############
+function symLinkFiles {
+
+   if [ ! -L $ORACLE_BASE_CONFIG/dbs/spfile$ORACLE_SID.ora ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/spfile$ORACLE_SID.ora $ORACLE_BASE_CONFIG/dbs/spfile$ORACLE_SID.ora
+   fi;
+   
+   if [ ! -L $ORACLE_BASE_CONFIG/dbs/orapw$ORACLE_SID ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/orapw$ORACLE_SID $ORACLE_BASE_CONFIG/dbs/orapw$ORACLE_SID
+   fi;
+   
+   if [ ! -L $ORACLE_BASE_HOME/network/admin/sqlnet.ora ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/sqlnet.ora $ORACLE_BASE_HOME/network/admin/sqlnet.ora
+   fi;
+
+   if [ ! -L $ORACLE_BASE_HOME/network/admin/listener.ora ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/listener.ora $ORACLE_BASE_HOME/network/admin/listener.ora
+   fi;
+
+   if [ ! -L $ORACLE_BASE_HOME/network/admin/tnsnames.ora ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/tnsnames.ora $ORACLE_BASE_HOME/network/admin/tnsnames.ora
+   fi;
+
+   # oracle user does not have permissions in /etc, hence cp and not ln 
+   cp $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/oratab /etc/oratab
+
+}
+
+########### SIGINT handler ############
+function _int() {
+   echo "Stopping container."
+   echo "SIGINT received, shutting down database!"
+   sqlplus / as sysdba <<EOF
+   shutdown immediate;
+   exit;
+EOF
+   lsnrctl stop
+}
+
+########### SIGTERM handler ############
+function _term() {
+   echo "Stopping container."
+   echo "SIGTERM received, shutting down database!"
+   sqlplus / as sysdba <<EOF
+   shutdown immediate;
+   exit;
+EOF
+   lsnrctl stop
+}
+
+###################################
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! #
+############# MAIN ################
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! #
+###################################
+
+# Check whether container has enough memory
+if [[ -f /sys/fs/cgroup/cgroup.controllers ]]; then
+  memory=$(cat /sys/fs/cgroup/memory.max)
+else
+  memory=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)
+fi
+
+# Github issue #219: Prevent integer overflow,
+# only check if memory digits are less than 11 (single GB range and below) 
+if [[ ${memory} != "max" && ${#memory} -lt 11 && ${memory} -lt 8589934592 ]]; then
+    echo "Error: The container doesn't have enough memory allocated."
+    echo "A database container needs at least 8 GB of memory."
```

Comment:
This is quite an increase from 2GB. Do we really need 8GB?
Lots of people still only have 8GB in total in their laptops, this would prevent them from ever running the image locally.

Reply:
Sorry, my fault ! 2GB memory is enough for 21c. I put 8GB there by mistake. Corrected in the commit 82b7492dd9615f56a7696bf04f26674ba907578d.